ffmpeg/ffmpeg_test.go

```diff
@@ -1037,3 +1048,292 @@ func TestTranscoder_StreamCopyAndDrop(t *testing.T) {
+	out := []TranscodeOptions{TranscodeOptions{Oname: dir + "/0.ts", Profile: P144p30fps16x9}}
+	res, err := Transcode3(in, out)
+	if err != nil || res.Decoded.Frames != 62 || res.Encoded[0].Frames != 31 {
+		t.Error("Unexpected preconditions ", err, res)
+	}
+	frames := res.Encoded[0].Frames
+
+	// Transcode results repeatedly, ensuring we have the same results each time
+	for i := 0; i < 5; i++ {
+		in.Fname = fmt.Sprintf("%s/%d.ts", dir, i)
+		out[0].Oname = fmt.Sprintf("%s/%d.ts", dir, i+1)
+		res, err = Transcode3(in, out)
+		if err != nil ||
+			res.Decoded.Frames != frames || res.Encoded[0].Frames != frames {
+			t.Error(fmt.Sprintf("Unexpected frame count for input %d : decoded %d encoded %d", i, res.Decoded.Frames, res.Encoded[0].Frames))
+		}
+		if res.Decoded.Frames != 31 && res.Encoded[0].Frames != 31 {
+			t.Error("Unexpected frame count! ", res)
+		}
+	}
+
+	// Do the same but with audio. This yields a 30fps file.
+	in = &TranscodeOptionsIn{Fname: dir + "/test-short-with-audio.ts"}
+	out = []TranscodeOptions{TranscodeOptions{Oname: dir + "/audio-0.ts", Profile: P144p30fps16x9}}
+	res, err = Transcode3(in, out)
+	if err != nil || res.Decoded.Frames != 60 || res.Encoded[0].Frames != 30 {
+		t.Error("Unexpected preconditions ", err, res)
+	}
+	frames = res.Encoded[0].Frames
+
+	// Transcode results repeatedly, ensuring we have the same results each time
+	for i := 0; i < 5; i++ {
+		in.Fname = fmt.Sprintf("%s/audio-%d.ts", dir, i)
+		out[0].Oname = fmt.Sprintf("%s/audio-%d.ts", dir, i+1)
+		res, err = Transcode3(in, out)
+		if err != nil ||
+			res.Decoded.Frames != frames || res.Encoded[0].Frames != frames {
+			t.Error(fmt.Sprintf("Unexpected frame count for input %d : decoded %d encoded %d", i, res.Decoded.Frames, res.Encoded[0].Frames))
+		}
+	}
+}
+
+func TestTranscoder_MismatchedEncodeDecode(t *testing.T) {
+	// Encoded frame count does not match decoded frame count for mp4
+	// Note this is not an issue for mpegts! (this is sanity checked)
+
+	run, dir := setupTest(t)
+	defer os.RemoveAll(dir)
+
+	p144p60fps := P144p30fps16x9
+	p144p60fps.Framerate = 60
+
+	cmd := `
+        set -eux
+        cd $0
+
+        # prepare 1-second input
+        cp "$1/../transcoder/test.ts" inp.ts
+        ffmpeg -loglevel warning -i inp.ts -c:a copy -c:v copy -t 1 test.ts
+    `
+	run(cmd)
+
+	in := &TranscodeOptionsIn{Fname: dir + "/test.ts"}
+	out := []TranscodeOptions{TranscodeOptions{Oname: dir + "/out.mp4", Profile: p144p60fps}}
+	res, err := Transcode3(in, out)
+	if err != nil {
+		t.Error(err)
+	}
+	in.Fname = dir + "/out.mp4"
+	res2, err := Transcode3(in, nil)
+	if err != nil {
+		t.Error(err)
+	}
+	// TODO Ideally these two should match. As far as I can tell it is due
+	//      to timestamp rounding around EOF. Note this does not happen with
+	//      mpegts formatted output!
+	if res2.Decoded.Frames != 60 || res.Encoded[0].Frames != 61 {
+		t.Error("Did not get expected frame counts: is this fixed?!? ",
```

Comment:
do we need the "is this fixed?!?" bit?

Reply:
Happy to make adjustments to the language if you can think of a better phrasing :) Mostly, the intent was to express some notion of:

* This behavior is currently broken, and this unit test serves as reproducible documentation (along with https://github.com/livepeer/lpms/issues/155)
* But we need to make broken tests "pass"
* If the broken test no longer passes, maybe it means the underlying issue has been fixed? 🤞 
* If the test breaks, hopefully a distinct error message will prompt more careful review of the issue and any associated changes.

Reply:
ok, that makes a lot more sense now - thank you for explaining :-)

Reply:
Updated the language a bit in https://github.com/livepeer/lpms/pull/153/commits/ab98e5177bf328322e594a3efdc86a0dc3225ab2